### DISCUSSION TITLE
Installation on Mac M1 Arm64

### DISCUSSION THREAD

**Discussion by xiaodi0303** [SSCHAcode/python-sscha#141] [Q&A] [answered]
Hi SSCHA Developers, 

Hope you are doing great. When I'm trying to install CellConstructor and python-sscha libraries, I finished the job with pip. But when I ran the `cellconstructor_test.py`, I hit a problem raised from `import symph`, which noticed me that `symph.cpython-39-darwin.so` is an incompatible structure to Arm64. 

Is there any good advices for Mac M1 users? Thank you very much! 

Best regards, 
Chendi Xie
Dept of Phys & Astro
Clemson Univ

**Comment by mesonepigreco** [comment]
Hi, 
I just uploaded the new guide on the sscha webpage to install with anaconda:
https://sscha.eu/download/

Try to see if the anaconda installation works on the M1 machine.

**Reply by xiaodi0303** [reply]
Hi! 

When I go to the last line `pip install cellconstructor python-sscha tdscha`, it shows me an extremely long error script as below. 

![sscha_error.txt](https://github.com/SSCHAcode/python-sscha/files/12137529/sscha_error.txt)

Could you please have a quick look if there's an error in my installation?Thank you! 

Best regards, 
Chendi Xie
Dept of Physics & Astronomy
Clemson University

**Reply by mesonepigreco** [reply]
It seems that Python fails to recognize the Fortran compiler.
Can you try to run the following command?

```
CC=gcc FC=gfortran LDSHARED=gfortran pip install cellconstructor python-sscha tdscha
```

**Reply by yjiang-iop** [reply]
Hi SSCHA Developers!

I am using an M2 Macbook and also run into the same problem. The command you give above does not work either. Do you have any other suggestions for it? or should I use a virtual machine? Thank you in advance!

Best,
Yi Jiang
DIPC

**Reply by xiaodi0303** [reply]
> It seems that Python fails to recognize the Fortran compiler.
> Can you try to run the following command?
> 
> ```
> CC=gcc FC=gfortran LDSHARED=gfortran pip install cellconstructor python-sscha tdscha
> ```
> 
> 

Hi, 

Thank you for your quick response. I tried the new command line and below is the error message. 

![1_sscha_error.txt](https://github.com/SSCHAcode/python-sscha/files/12159628/1_sscha_error.txt)

This seems quite similar to the former one. Could you please have a quick look? Thank you again! 

Best regards, 
Chendi Xie
Dept of Physics and Astronomy
Clemson University

**Reply by mesonepigreco** [reply]
So it seems that python errors because of a warning that the fortran compiler raises while python tries to understand the version of the compiler:

```
ValueError: A valid Fortran version was not found in this string:
            x86_64-apple-darwin13.4.0-gfortran: warning: could not understand version 13.05.00
            11.3.0
```

The Fortran version is correctly printed in output, but python analyzes also the warning, failing to extract the last line.
I will have the possibility to work on a M1 apple machine during August and come up with a solution.

In the meantime, you can setup a virtual machine with linux on it (we tested it for the SSCHA school and works on Apple M1 and M2 computers).

**Reply by mesonepigreco** [reply, marked as answer]
The virtual machine that works on M1 and M2 Apple machines with the sscha code installed can be found here:
https://sites.google.com/view/hubbard-koopmans-2023/quantum-mobile

There is a specific installation guide for the virtual machine that works on the new ARM architectures (M* CPUs)

To use the SSCHA within this quantum mobile, activate the environment with

```bash
conda activate sscha
```
Inside a terminal before executing the Python scripts that load the SSCHA.

**Comment by mesonepigreco** [comment]
@ionerrea  
Do you manage to install the SSCHA code directly on the M1 or did you use the virtual machine?
